WireAuthentication/Sources/WireAuthenticationLogic/DetermineAuthMethodUseCase.swift

```diff
@@ -93,11 +93,35 @@ package struct DetermineAuthMethodUseCase: DetermineAuthMethodUseCaseProtocol {
             return .loginViaSSO(code: ssoCode)
 
         case .backend:
-            guard let backendURL = configuration.backendURL else {
+            guard let configURL = configuration.backendURL else {
                 throw AuthenticationAPIError.invalidResponse
             }
+            let backendURL = try await fetchBackendConfig(from: configURL)
+
             return .onPremLogin(email: email, backendConfig: backendURL)
         }
     }
 
+    private func fetchBackendConfig(from backendURL: URL) async throws -> URL {
+        let (data, _) = try await URLSession.shared.data(from: backendURL)
+
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
```

Comment:
question: is this needed if we provide explicit coding keys?

Reply:
No, we don't need because I added `CodingKeys`